lib/email-addresses.js

```diff
@@ -833,34 +878,60 @@ function parse5322(opts) {
     if (opts === null) { return null; }
 
     parseString = opts.input;
+    
+    var startAtMap = {
+        'from': fromSpec,
+        'sender': senderSpec,
+        'reply-to': addressList,
+    };
+    
+    var giveResMap = {
+        'from': fromGiveResult,
+        'sender': senderGiveResult,
+        'reply-to': addrListGiveResult,
+    };
+    
+    // console.log("Starting at: " + opts.startAt);
+    
+    var startAt = startAtMap[opts.startAt] || addressList;
+    var giveResult = giveResMap[opts.startAt] || addrListGiveResult;
 
     if (!opts.strict) {
         initialize();
         opts.strict = true;
-        parsed = addressList(parseString);
+        parsed = startAt(parseString);
         if (opts.partial || !inStr()) {
             return giveResult(parsed);
         }
         opts.strict = false;
     }
 
     initialize();
-    parsed = addressList(parseString);
+    parsed = startAt(parseString);
     if (!opts.partial && inStr()) { return null; }
     return giveResult(parsed);
 }
 
-function parseOneAddressSimple(opts) {
-    var result;
-
+function _opts (opts, startAt) {
```

Comment:
Merge this with `handleOpts` since it's not otherwise clear why there are two functions `_opts` and `handleOpts`.